Makefile

```diff
@@ -1,85 +1,92 @@
-# Copyright 2016 The Kubernetes Authors.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-include ./hack/Makefile.mk
-
-# Directories that the make will recurse into.
-DIRS := \
-  contrib/registry \
-  contrib/broker/k8s \
-  contrib/broker/user_provided \
-  contrib/broker/server \
-  pkg \
-  pkg/apis/servicecatalog \
-  pkg/apis/servicecatalog/v1alpha1 \
-  pkg/controller/catalog \
-  cmd/service-catalog
-
-ALL := all build build-linux build-darwin clean docker push test lint coverage
-SUB := $(addsuffix .sub, $(ALL))
-.PHONY: $(ALL) $(SUB) format
-
-# Recursive targets.
-build: build.sub
-build-linux: build-linux.sub
-build-darwin: build-darwin.sub
-docker: docker.sub
-lint: lint.sub
-push: push.sub
-test: test.sub
-
-clean: clean.sub
-	rm -rf $(BINDIR)
-	rm -f .dockerInit
-	rm -f .scBuildImage
-	docker rmi -f scbuildimage > /dev/null 2>&1 || true
+all: build test verify
```

Comment:
@duglin can you put a TODO list in the PR description? something like this:

TODO:

- [ ] remove `hack/*.mk`
- [ ] remove/migrate bits from `BUILD_SYSTEM.md` according to new build logic
- [ ] ...